### PR TITLE
PMM-10913-server-with-external-postgres-server

### PR DESCRIPTION
**Pull request by rishat-ishbulatov**
[PMM-10913](https://jira.percona.com/browse/PMM-10913)

Build: [PMM-10913](https://github.com/Percona-Lab/pmm-submodules/pull/2965)

### CODE REVIEW

File: docs/details/postgresql.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+PMM predefines certain flags that allow you to use PostgreSQL parameters as environment variables:
+
+!!! caution alert alert-warning "Warning"
+     The `POSTGRES_*` environment variables are experimental and subject to change. It is recommended that you use these variables for testing purposes only and not on production.
+
+To use PostgreSQL as an external database instance, use the following environment variables: 
+ 
+`POSTGRES_ADDR` -> [postgres-addr](https://www.postgresql.org/docs/14/libpq-connect.html#LIBPQ-CONNECT-HOST)
+:   Host name and port for external PostgreSQL database.
```
Comment (denisok) [inline docs/details/postgresql.md:15]:
```suggestion
:   Hostname and port for external PostgreSQL database.
```

**Review by denisok — CHANGES_REQUESTED** [review]

File: docs/details/postgresql.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+`POSTGRES_SSL_CA_PATH` -> [postgres-ssl-ca-path](https://www.postgresql.org/docs/14/libpq-connect.html#LIBPQ-CONNECT-SSLROOTCERT)
+:   This parameter specifies the name of a file containing SSL certificate authority (CA) certificate(s).
+
+`POSTGRES_SSL_KEY_PATH` -> [postgres-ssl-key-path](https://www.postgresql.org/docs/14/libpq-connect.html#LIBPQ-CONNECT-SSLKEY)
+:   This parameter specifies the location for the secret key used for the client certificate.
+
+`POSTGRES_SSL_CERT_PATH` -> [postgres-ssl-cert-path](https://www.postgresql.org/docs/14/libpq-connect.html#LIBPQ-CONNECT-SSLCERT)
+:   This parameter specifies the file name of the client SSL certificate.
+
+By default communications between server and database are not encrypted. In order to secure connection you can read [this](https://www.postgresql.org/docs/14/ssl-tcp.html) article and provide `POSTGRES_SSL_*` variables.
```
[the remark this answers is not in the document]
Comment (rasika-chivate) [reply docs/details/postgresql.md:40]: " Communication between the server and the database is not encrypted by default. To secure a connection, follow [PostgreSQL SSL instructions](https://www.postgresql.org/docs/14/ssl-tcp.html) and provide the POSTGRES_SSL_*` variables."

File: docs/details/postgresql.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+By default communications between server and database are not encrypted. In order to secure connection you can read [this](https://www.postgresql.org/docs/14/ssl-tcp.html) article and provide `POSTGRES_SSL_*` variables. 
+
+**Example**
+
+To use PostgreSQL as an external database: 
+
+* Generate all nessesary SSL certificates.
+* Build Percona Server with certificates under read-only rights and  grafana user and group.
+* Build PostgreSQL image with pg_hba.conf and certificates.
```
[the remark this answers is not in the document]
Comment (rasika-chivate) [reply docs/details/postgresql.md:48]: Would attach be the appropriate word?

File: docs/details/postgresql.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+:   This parameter specifies the file name of the client SSL certificate.
+
+By default communications between server and database are not encrypted. In order to secure connection you can read [this](https://www.postgresql.org/docs/14/ssl-tcp.html) article and provide `POSTGRES_SSL_*` variables. 
+
+**Example**
+
+To use PostgreSQL as an external database: 
+
+* Generate all nessesary SSL certificates.
+* Build Percona Server with certificates under read-only rights and  grafana user and group.
```
[the remark this answers is not in the document]
Comment (rasika-chivate) [reply docs/details/postgresql.md]: "Deploy Percona Server with certificates under read-only permissions and `grafana` user and grafana group."

File: docs/details/postgresql.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+* Generate all nessesary SSL certificates.
+* Deploy Percona Server with certificates under read-only permissions and grafana user and grafana group.
+* Attach pg_hba.conf and certificates to the PostgreSQL image.
+* Run PostgreSQL server.
+```sh
+docker run 
+--name external-postgres 
+-e POSTGRES_PASSWORD=secret 
+<image_id>
```
Comment (denisok) [inline docs/details/postgresql.md:40]: I think we don't support any Percona PG versions?

We probably need specific version here of Percona PG

File: docs/details/postgresql.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+--name percona-server 
+-e POSTGRES_ADDR=$ADDRESS:$PORT
+-e POSTGRES_DBNAME=$DBNAME
+-e POSTGRES_USERNAME=$USER
+-e POSTGRES_DBPASSWORD=$PASSWORD
+-e POSTGRES_SSL_MODE=$SSL_MODE
+-e POSTGRES_SSL_CA_PATH=$CA_PATH
+-e POSTGRES_SSL_KEY_PATH=$KEY_PATH
+-e POSTGRES_SSL_CERT_PATH=$CERT_PATH 
+<image_id>
```
Comment (denisok) [inline docs/details/postgresql.md:64]:
```suggestion
percona/pmm-server:2
```

**Review by denisok — APPROVED** [review]

File: docs/details/postgresql.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+| `POSTGRES_SSL_CA_PATH`    | [postgres-ssl-ca-path](https://www.postgresql.org/docs/14/libpq-connect.html#LIBPQ-CONNECT-SSLROOTCERT)      | This parameter specifies the name of a file containing SSL certificate authority (CA) certificate(s).
+| `POSTGRES_SSL_KEY_PATH`    | [postgres-ssl-key-path](https://www.postgresql.org/docs/14/libpq-connect.html#LIBPQ-CONNECT-SSLKEY)      | This parameter specifies the location for the secret key used for the client certificate.
+| `POSTGRES_SSL_CERT_PATH`    | [postgres-ssl-cert-path](https://www.postgresql.org/docs/14/libpq-connect.html#LIBPQ-CONNECT-SSLCERT)     | This parameter specifies the file name of the client SSL certificate.
+
+
+By default, communications between the PMM server and the database are not encrypted. In order to secure a connection, please follow [PostgeSQL SSL instructions](https://www.postgresql.org/docs/14/ssl-tcp.html) and provide `POSTGRES_SSL_*` variables.
```
Comment (rasika-chivate) [inline docs/details/postgresql.md:26]:
```suggestion
By default, communication between the PMM server and the database is not encrypted. To secure a connection, follow [PostgeSQL SSL instructions](https://www.postgresql.org/docs/14/ssl-tcp.html) and provide `POSTGRES_SSL_*` variables. 
```

File: docs/details/postgresql.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+| `POSTGRES_SSL_CERT_PATH`    | [postgres-ssl-cert-path](https://www.postgresql.org/docs/14/libpq-connect.html#LIBPQ-CONNECT-SSLCERT)     | This parameter specifies the file name of the client SSL certificate.
+
+
+By default, communications between the PMM server and the database are not encrypted. In order to secure a connection, please follow [PostgeSQL SSL instructions](https://www.postgresql.org/docs/14/ssl-tcp.html) and provide `POSTGRES_SSL_*` variables. 
+
+**Example**
+
+To use PostgreSQL as an external database: 
+
+* Generate all nessesary SSL certificates.
```
Comment (rasika-chivate) [inline docs/details/postgresql.md:32]:
```suggestion
1. Generate all necessary SSL certificates.
```

File: docs/details/postgresql.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+* Deploy Percona Server with certificates under read-only permissions and grafana user and grafana group.
```
Comment (rasika-chivate) [inline docs/details/postgresql.md:33]:
```suggestion
2. Deploy Percona Server with certificates under read-only permissions and Grafana user and Grafana group.
```

File: docs/details/postgresql.md [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+* Attach pg_hba.conf and certificates to the PostgreSQL image.
+* Run PostgreSQL server.
```
Comment (rasika-chivate) [inline docs/details/postgresql.md:35]:
```suggestion
4. Run PostgreSQL server.
```

**Review by rasika-chivate — CHANGES_REQUESTED** [review]

File: docs/details/postgresql.md (hunk 5 above)
Comment (rishat-ishbulatov) [reply docs/details/postgresql.md:40]: minimum required PG version warning added

**Review by rasika-chivate — CHANGES_REQUESTED** [review]

**Pull request merged**